src/VisualStudio/TestUtilities/Workspace/Solution.cs

```diff
@@ -45,13 +45,27 @@ public string FileName
             }
         }
 
+        public void Save()
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(FileName));
+            DteSolution.SaveAs(FileName);
+        }
+
+        public Project OpenProject(string projectFilePath)
+        {
+            var dteProject = _dteSolution.AddFromFile(projectFilePath);
+            Save();
+            return new Project(dteProject, this, ProjectLanguage.CSharp);
+        }
+
         public Project AddProject(string projectName, ProjectTemplate projectTemplate, ProjectLanguage projectLanguage)
         {
             var solutionFolder = Path.GetDirectoryName(FileName);
-            var projectPath = Path.Combine(solutionFolder, projectName);
+            var projectPath = Path.Combine(solutionFolder, Path.GetFileNameWithoutExtension(projectName));
```

Comment:
It would likely provide a better experience by appending the extension below based on the project language we are given (I believe it would also match the current TAO behavior, which is desired where possible).